File: in-motion-server/Friends/IMS.Friends.Models/Dto/Outgoing/RequestFriendshipDto.cs
namespace IMS.Friends.Models.Dto.Outgoing;

public class RequestFriendshipDto
{
    public string Id { get; set; }
    
    public string ExternalUserId { get; set; }
    
    public FriendInfoDto ExternalUser { get; set; }
    
    public DateTime Requested { get; set; }
}

File: in-motion-android/app/src/main/java/com/inmotion/in_motion_android/data/remote/dto/friends/RequestFriendshipDto.kt
package com.inmotion.in_motion_android.data.remote.dto.friends

import com.inmotion.in_motion_android.data.database.entity.RequestedFriend

data class RequestFriendshipDto(
    val id: String,
    val externalUserId: String,
    val externalUser: FriendInfoDto,
    val requested: String
) {
    fun toRequestedFriend(): RequestedFriend {
        return RequestedFriend(
            externalUserId,
            id,
            externalUser.nickname,
            externalUser.bio,
            requested
        )
    }
}
